### PR TITLE
fix: race conditions in tests

### PR DESCRIPTION
**Pull request by stevenh**
Fix the race conditions in test and simplify the use of newServer.

### CODE REVIEW

**Comment by stevenh** [comment]
Seems like there's still something funky going on with server teardown, needs more investigation.

File: mockserver_test.go [hunk 1]
```diff
@@ -307,22 +357,43 @@ func newSSHServerShell(conn net.Conn) (*sshServerShell, error) {
...
+}
+
+func (c *sshServerShell) waitChannel() (ssh.Channel, error) {
+	ch, closed := c.channel()
+	for ch == nil {
+		if closed {
+			return nil, net.ErrClosed
+		}
 		c.cond.Wait()
+		ch, closed = c.channel()
```
Comment (HalloTschuess) [inline mockserver_test.go:383]: There should probably be another check whether `closed` is true. Theoretically (even if unlikely) there could be a `channel` which is already closed.

Another possibility is to return `nil` from `c.channel()` if closed.

**Review by HalloTschuess — CHANGES_REQUESTED** [review]
Thanks @stevenh for fixing and improving the test server. I totally failed to see these data races. I just checked the previous version with `go test -race ./...` and it actually deteced them. Now this is no longer the case.  
I left some minor comments.

File: basic_cmds_test.go [hunk 2]
```diff
@@ -48,7 +41,8 @@ func TestCmdsBasicSSH(t *testing.T) {
 	testCmdsBasic(t, c)
 }
 
-func testCmdsBasic(t *testing.T, c *Client) { //nolint: thelper
+func testCmdsBasic(t *testing.T, c *Client) {
+	t.Helper()
```
[the remark this answers is not in the document]
Comment (stevenh) [reply basic_cmds_test.go:45]: When a function is marked as a helper you get an entry in the reported stack, so it's helpful either way. So it's always helpful to do this.

[the remark this answers is not in the document]
Comment (stevenh) [reply basic_cmds_test.go:45]: Quick test with `auth`  failing on its first line shows with `t.Helper()` you get:
```
=== RUN   TestCmdsBasic/auth
    basic_cmds_test.go:24:
                Error Trace:    basic_cmds_test.go:48
                Error:          Should be true
                Test:           TestCmdsBasic/auth
```
Without it you get:
```
=== RUN   TestCmdsBasic
=== RUN   TestCmdsBasic/auth
    basic_cmds_test.go:116:
                Error Trace:    basic_cmds_test.go:48
                Error:          Should be true
                Test:           TestCmdsBasic/auth
--- FAIL: TestCmdsBasic (0.00s)
    --- FAIL: TestCmdsBasic/auth (0.00s)
FAIL
exit status 1
FAIL    github.com/multiplay/go-ts3     0.009s
```

I would expect the former as I want to know which test triggered the error and that is `TestCmdsBasic` and not `t.Run(tc.name, tc.f)` within `testCmdsBasic`.

File: server_cmds_test.go [hunk 3]
```diff
@@ -50,7 +43,8 @@ func TestCmdsServerSSH(t *testing.T) {
 	testCmdsServer(t, c)
 }
 
-func testCmdsServer(t *testing.T, c *Client) { //nolint: thelper
+func testCmdsServer(t *testing.T, c *Client) {
+	t.Helper()
```
[the remark this answers is not in the document]
Comment (stevenh) [reply server_cmds_test.go:47]: Answer above.

File: mockserver_test.go (hunk 1 above)
Comment (stevenh) [reply mockserver_test.go]: Nice catch, fixed.

Also triggered another race when testing this fix, so there's additional validation to ignore `connection reset by peer on close`

File: basic_cmds_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (HalloTschuess) [reply basic_cmds_test.go:45]: Hm yeah makes sense. I didn't think about testify showing the `Error Trace`. Also you are right about `t.Run`. I just remenbered having some problems with it back in the day when I didn't use testify yet. Looks good then.

[the remark this answers is not in the document]
Comment (stevenh) [reply basic_cmds_test.go:45]: Before sub tests as well 👍

**Review by HalloTschuess — APPROVED** [review]
LGTM

**Pull request merged**